lib/active_support/cache/memcached_store.rb

```diff
@@ -168,6 +178,17 @@ def cas_multi(*names, **options)
 
               Hash[serialized_values]
             end
+
+            if local_cache
+              keys_to_names.each_key do |key|
```

Comment:
This is iterating through the fetched keys, but it should instead iterate through the keys that it attempted to update, which would be the keys for the hash returned from the `@connection.cas` block (`Hash[serialized_values]`).

Reply:
Good catch.